### PR TITLE
PoC: darwin.apple_sdk_11_0.gcc12Stdenv: init

### PR DESCRIPTION
**Pull request by wegank**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build signalbackup-tools

**Review by malob — APPROVED** [review]
LGTM. Thanks!

**Comment by malob** [comment]
@wegank, @mweinelt, what's the process for pulling the trigger on merging this PR? Is there anything I can do to help?

**Comment by mweinelt** [comment]
I'll defer to @NixOS/darwin-maintainers in this case.

File: pkgs/os-specific/darwin/apple-sdk-11.0/default.nix [hunk 1]
```diff
@@ -56,6 +56,21 @@ let
     '';
   };
 
+  mkClangStdenv = stdenv: let
```
Comment (winterqt) [inline pkgs/os-specific/darwin/apple-sdk-11.0/default.nix:59]: Isn't the fact that this is Clang kind of superfluous, as it just uses the cc from the provided stdenv?

**Review by winterqt — APPROVED** [review]
LGTM, one nit though.

File: pkgs/os-specific/darwin/apple-sdk-11.0/default.nix (hunk 1 above)
Comment (Et7f3) [reply pkgs/os-specific/darwin/apple-sdk-11.0/default.nix:59]: We can use another version of clang ?

Comment (winterqt) [reply pkgs/os-specific/darwin/apple-sdk-11.0/default.nix:59]: I don't understand your question, but to clarify, I'm talking about solely the function name here.

Comment (wegank) [reply pkgs/os-specific/darwin/apple-sdk-11.0/default.nix]: Renamed to `mkStdenv`. Feel free to suggest a better (or more endomorphic?) name.

**Review by amz-x — APPROVED** [review]

**Comment by winterqt** [comment]
So does the latest version of signalbackup-tools not work with Clang at all?

**Comment by winterqt** [comment]
> Nope, it doesn't work with `clang14Stdenv` or `clang15Stdenv`, and that's why I simply marked it as broken in #215791.

That PR is why I asked :)

Also: why? Would be nice if you added logs or a summary or something in at least the PR/commit description. (Or a comment next to the `meta.broken` deceleration).

> It does work with `gcc12Stdenv`, however, but I haven't tested it on x86_64-darwin.

Test with Rosetta?

**Comment by wegank** [comment]
And `darwin.apple_sdk_11_0.gcc12Stdenv` works, ironically...

Also for the error under `clang15Stdenv`:

```bash
In file included from signalbackup/htmlprepbody.cc:20:
In file included from signalbackup/signalbackup.ih:20:
In file included from signalbackup/signalbackup.h:23:
In file included from signalbackup/../sqlitedb/sqlitedb.h:25:
In file included from /nix/store/54g2wsly5nqn640f7yyjab81fc4vn61s-libcxx-15.0.7-dev/include/c++/v1/vector:275:
In file included from /nix/store/54g2wsly5nqn640f7yyjab81fc4vn61s-libcxx-15.0.7-dev/include/c++/v1/__algorithm/equal.h:13:
/nix/store/54g2wsly5nqn640f7yyjab81fc4vn61s-libcxx-15.0.7-dev/include/c++/v1/__algorithm/comp.h:73:71: error: invalid operands to binary expression ('const SignalBackup::Range' and 'const SignalBackup::Range')
    bool operator()(const _T1& __x, const _T1& __y) const {return __x < __y;}
```

**Comment by winterqt** [comment]
Hmm, maybe worth filing upstream?

Also:
1. Maybe this should be its own PR, as technically the scope (e.g. branch name) has changed? (If the initial branch name was just focused on fixing the broken package, it wouldn't matter, but in this case...)
2. Either way, you should probably mention why we're adding the GCC stdenv in the commit message (and why it was broken until now in the unbreak one?)

Again, I'm probably being way too pedantic here. Sorry 😕

**Comment by wegank** [comment]
Superseded by #219431.

**Pull request closed without merging**